models/service.go

```diff
@@ -30,36 +35,50 @@ type Service struct {
 	Autoscalers         Autoscalers         `json:"autoscalers"`
 }
 
-func GetServicesByNamespace(namespaceName string) ([]ServiceOverview, error) {
-	istioClient, err := kubernetes.NewClient()
-	if err != nil {
-		return nil, err
-	}
+func (s *ServiceList) SetServiceList(serviceList *kubernetes.ServiceList) {
+	s.Services = CastServiceOverviewCollection(serviceList)
+}
 
-	services, err := istioClient.GetServices(namespaceName)
-	if err != nil {
-		return nil, err
+func CastServiceOverviewCollection(sl *kubernetes.ServiceList) []ServiceOverview {
+	if sl.Services == nil {
+		return nil
 	}
 
-	return CastServiceOverviewCollection(services), nil
-}
+	services := make([]ServiceOverview, len(sl.Services.Items))
 
-func CastServiceOverviewCollection(sl *v1.ServiceList) []ServiceOverview {
-	services := make([]ServiceOverview, len(sl.Items))
-	for i, item := range sl.Items {
-		services[i] = CastServiceOverview(item)
+	for i, item := range sl.Services.Items {
+		services[i] = CastServiceOverview(&item, sl.Deployments)
 	}
 
 	return services
 }
 
-func CastServiceOverview(s v1.Service) ServiceOverview {
+func CastServiceOverview(s *v1.Service, deployments *v1beta1.DeploymentList) ServiceOverview {
 	service := ServiceOverview{}
 	service.Name = s.Name
 
+	replicas, availableReplicas, unavailableReplicas := getPodStatusForService(s.Labels[config.Get().ServiceFilterLabelName], deployments)
+	service.Replicas = replicas
+	service.AvailableReplicas = availableReplicas
+	service.UnavailableReplicas = unavailableReplicas
+
 	return service
 }
 
+func getPodStatusForService(serviceName string, deployments *v1beta1.DeploymentList) (int32, int32, int32) {
+	replicas, availableReplicas, unavailableReplicas := int32(0), int32(0), int32(0)
+
+	for _, deployment := range deployments.Items {
+		if deployment.ObjectMeta.Labels != nil && deployment.ObjectMeta.Labels[config.Get().ServiceFilterLabelName] == serviceName {
+			replicas = replicas + deployment.Status.Replicas
+			availableReplicas = availableReplicas + deployment.Status.AvailableReplicas
```

Comment:
Ok, just to confirm that I understood it, it accumulates the pods available and unavailable data of all versions (v1, v2, v3, vn) of a service, and shown in the list.

Reply:
yes, it is. this way, we can rapidly see from service list which services need special attention.
Thanks for the review :dancer: